### ISSUE TITLE
Errors when editing columns in Vizy blocks

### ISSUE THREAD

**Issue by pl-mnm** [verbb/tablemaker#28]
Hi, I'm facing a couple of issues when I insert a Table Maker field as part of a Vizy block:

- The `rows` table doesn't get updated when new columns are created unless I save the entry
- When I delete columns, I get the following error:
```
PHP Notice – yii\base\ErrorException
Undefined offset: 3
```
In the CP, I get the following console error comming from `tablemaker.js:247:17`:
```
Uncaught TypeError: can't access property "block", columns is undefined
```

I'm currently on Craft 3.7.25.1 and using Table Maker 2.0.1 and Vizy 1.0.9.

Thanks in advance.

**Comment by anchovy** [comment]
I've got the same issue when used as a vizy block.
Adding columns doesn't automatically create the table content row below. But if you add the column save and go back into the entry the row table content row is generated.

Error I get is:
```
tablemaker.js:158 Uncaught TypeError: Cannot read properties of undefined (reading 'block')
    at s.constructor.getDataFromTables (tablemaker.js:158:27)
    at s.constructor.reconstructRowsTable (tablemaker.js:109:14)
    at s.constructor.<anonymous> (Base.js:153:23)
    at HTMLTextAreaElement.o (jquery.js?v=1659503183:2:89240)
    at HTMLTextAreaElement.handle (Garnish.js:946:34)
    at HTMLTextAreaElement.dispatch (jquery.js?v=1659503183:2:43392)
    at v.handle (jquery.js?v=1659503183:2:41387)
    at Object.trigger (jquery.js?v=1659503183:2:71930)
    at HTMLTextAreaElement.<anonymous> (jquery.js?v=1659503183:2:72526)
    at Function.each (jquery.js?v=1659503183:2:3002)
```

Craft version: Craft Pro 3.7.50
Table Maker: 3.0.3
Vizy: 1.0.17

**Comment by engram-design** [comment]
Fixed in [3.0.4](https://github.com/verbb/tablemaker/releases/tag/3.0.4)
